packages/flutter/lib/src/material/dialog.dart

```diff
@@ -219,6 +219,61 @@ class Dialog extends StatelessWidget {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=stateful_widget_scaffold_center}
+///
+/// ```dart
+/// @override
+/// Widget build(BuildContext context) {
+///   Future<void> _showMyDialog() async {
```

Comment:
Wouldn't the future be a `String`, not void?
```suggestion
///   Future<String> _showMyDialog() async {
```